File: src/main/java/voidchess/player/ki/concurrent/ConcurrencyStrategyFactory.java
package voidchess.player.ki.concurrent;

import voidchess.ui.ComputerPlayerUI;

public class ConcurrencyStrategyFactory {
    public static ConcurrencyStrategy getConcurrencyStrategy(ComputerPlayerUI ui, int numberOfCoresToUse) {
        if (numberOfCoresToUse == 1) {
            return new SingleThreadStrategy(ui);
        } else {
            return new MultiThreadStrategy(ui, numberOfCoresToUse);
        }
    }
}

File: src/main/kotlin/voidchess/player/ki/concurrent/ConcurrencyStrategyFactory.kt
package voidchess.player.ki.concurrent

fun getConcurrencyStrategy(showProgress: (Int, Int) -> Unit, numberOfCoresToUse: Int) = if (numberOfCoresToUse == 1) {
    SingleThreadStrategy(showProgress)
} else {
    MultiThreadStrategy(showProgress, numberOfCoresToUse)
}